lib/processor.js

```diff
@@ -16,12 +37,15 @@ const SUPPORTS_AUTOFIX = true;
 
 const markdown = unified().use(remarkParse);
 
+/**
+ * @type {Block[]}
+ */
 let blocks = [];
 
 /**
  * Performs a depth-first traversal of the Markdown AST.
  * @param {ASTNode} node A Markdown AST node.
- * @param {Object} callbacks A map of node types to callbacks.
+ * @param {{[key: string]: (node: ASTNode, parent: ASTParentNode) => void}} callbacks A map of node types to callbacks.
```

Comment:
```suggestion
 * @param {{[key: string]: (node: ASTNode) => void}} callbacks A map of node types to callbacks.
```

Reply:
I should also note that #177 added a fallback `*` callback for any node types that don't have their own callback. Right now, that callback doesn't need access to the AST node, so it isn't passed in the `else` case. If that complicates the type definition, feel free to get around that by passing `node` in the `else` case as well if that's the easiest solution.

https://github.com/eslint/eslint-plugin-markdown/blob/79be776331cf2bb4db2f265ee6cf7260e90e3d5e/lib/processor.js#L28-L32

https://github.com/eslint/eslint-plugin-markdown/blob/79be776331cf2bb4db2f265ee6cf7260e90e3d5e/lib/processor.js#L233-L235